### DISCUSSION TITLE
Viewing and opening account should be separated

### DISCUSSION THREAD

**Discussion by DejanMilicic** [akhansari/EsBankAccount#10] [Ideas]
The act of "opening" account as in "creation of new account" should exist as a separate event.
At the moment, "opening" account will not list unless you do a deposit or a withdrawal

**Comment by akhansari** [comment]
Good point, we should have `Opened` event. It doesn't make sens to have `Closed` without it.

I think I may have made a mistake from the beginning by choosing the traditional bank account domain:
[Why a bank account is not the best example of Event Sourcing?](https://event-driven.io/en/bank_account_event_sourcing/) \
But this topic is so easy, connects with everybody, can be done in 20mins during a kata and the UI is not complexe.

It might be worth changing to another domain that would be just as convenient.

**Comment by DejanMilicic** [comment]
I see nothing wrong with slowly pivoting towards a more feature-full example where you would introduce multitenancy, transfers between tenants, debit and credit cards, etc.
Banking is a domain everyone is familiar with.

Another option would be to "borrow" Mark Seeman's restaurant reservation system?

**Reply by akhansari** [reply]
Another example in Rust: https://github.com/thalo-rs/thalo/tree/main/examples/bank-account

**Comment by DejanMilicic** [comment]
This looks like similar domain but quite richer

https://github.com/atsapura/CardManagement

**Reply by akhansari** [reply]
Really interesting this repo 👍

**Comment by DejanMilicic** [comment]
There is also a very interesting project with an architecture that seems to be both very well structured and scalable 

https://github.com/kos59125/MvuArchitecture
